lib/libUPnP/Platinum/Source/Devices/MediaServer/PltMediaItem.cpp

```diff
@@ -528,7 +528,21 @@ PLT_MediaObject::ToDidl(NPT_UInt64 mask, NPT_String& didl)
             
             didl += " protocolInfo=\"";
             PLT_Didl::AppendXmlEscape(didl, m_Resources[i].m_ProtocolInfo.ToString());
-            didl += "\">";
+            didl += "\"";
+            /* adding custom data */
+            NPT_List<NPT_Map<NPT_String, NPT_String>::Entry*>::Iterator entry = m_Resources[i].m_CustomData.GetEntries().GetFirstItem();
+            while (entry)
+            {
+                didl += " ";
+                PLT_Didl::AppendXmlEscape(didl, (NPT_String&)(*entry)->GetKey());
```

Comment:
Is the explicit cast to `NPT_String&` necessary? `m_CustomData` already is a `NPT_Map<NPT_String, NPT_String>` so `GetKey()` and `GetValue()` should already return `NPT_String`.